packages/app/src/vis-packs/nexus/containers/NxImageContainer.tsx

```diff
@@ -33,8 +37,24 @@ function NxImageContainer(props: VisContainerProps) {
     keepRatio: guessKeepRatio(xAxisDef, yAxisDef),
   });
 
+  const nxDataToFetch =
+    selectedDef === signalDef
+      ? { ...nxData, auxDefs: [] } // don't fetch auxiliaries
+      : {
+          ...nxData,
+          signalDef: selectedDef, // fetch only selected auxiliary
+          auxDefs: [],
+          titleDataset: undefined,
+        };
```

Comment:
If I am not mistaken, this can be rewritten more simply:

```suggestion
  const nxDataToFetch =
       {
          ...nxData,
          signalDef: selectedDef,
          auxDefs: [],  // don't fetch auxiliaries
          ...(selectedDef !== signalDef ?  { titleDataset: undefined } : {})
        };
```

But why is it needed to conditionally set `titleDataset` ?

Reply:
Yep totally!

Regarding the title dataset, the idea is to show the auxiliary signal's label (and potentially `long_name`/`unit` attributes) as the title of the plot. If a `titleDataset` is always provided, then its value is used regardless of which signal is selected — I didn't think this was what we wanted.

Reply:
(If you agree with this behaviour, I'll add a comment to clarify.)

Reply:
Yeah all right.